### ISSUE TITLE
Renaming device does not cascade name to entities

### ISSUE THREAD

**Issue by ChevySSinSD** [simbaja/ha_gehome#127]
I have a washer, dryer, dishwasher, microwave, and oven working via this integration, thank you for all the work!  For all five devices, when I rename the device in the home assistant UI, I get a prompt asking if I should rename all entities, when I select "rename" none of the entities change (they all retain their default names).  I have confirmed the entity renaming works for devices not part of this integration.

**Comment by simbaja** [comment]
I've noticed that this has been hit or miss for me across a bunch of integrations, it seems like it's doing some sort of pattern matching to accomplish this.  I don't know exactly what the criteria are for this functionality (I didn't see any documentation on it), but if they can be determined, I can modify accordingly.

**Comment by regoras** [comment]
I did some digging and I think I figured out the reason this isn't working. The Device Name that the integration is defining is of format "GE (appliance type) (serial number)" while the entities for sensors and entities are getting named with format "(serial number) (entity/sensor name)". For the rename cascades to work correctly, I think (and wasn't able to figure out with my very rudimentary Python skills) the naming needs to match across the Device Name and the Entity/Sensor names. 

For example, my range device is named: GE Oven Serialnumber

And the entity for the oven is named: Serialnumber Oven
And a sensor for the oven is named: Serialnumber Display Temperature

I believe what would work is for the sensors and entities to match the device naming format.
